outputs/narrative.py

```
@@ -0,0 +1,124 @@
+    range_lbl = d0 if d0 == d1 else f"{d0} through {d1}"
+
+    total_h = sum(overall_days[d]["hours"] for d in days_sorted)
+    n_sess = sum(len(overall_days[d]["sessions"]) for d in days_sorted)
+    n_days = len(days_sorted)
+
+    window_start = dt_from.date().isoformat()
+    window_end = dt_to.date().isoformat()
+    if window_start == window_end == d0 == d1:
+        window_note = ""
+    elif window_start != window_end:
+        window_note = f" (report window {window_start}–{window_end})"
```

Ruff (0.15.9)

[warning] String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)